packages/common/src/types/field.ts

```diff
@@ -178,17 +178,16 @@ export enum TableCalculationType {
     BOOLEAN = 'boolean',
 }
 
-// TODO: this type and the TableCalculationField type share
-// most of the same fields -- can we merge them. This mostly
-// needs to add fieldType: FieldType.TABLE_CALCULATION which
-// would make the type checking more robust
 export type TableCalculation = {
     index?: number;
     name: string;
+    // This is a unique property of the table calculation
     displayName: string;
     sql: string;
     format?: CustomFormat;
     type?: TableCalculationType;
+    // TODO: Check if this needs to be required - it's not in the metricQuery compiled
+    fieldType?: FieldType.TABLE_CALCULATION;
```

Comment:
❓ I believe I should not add this here - it was only in `TableCalculationField`

Reply:
correct, we don't need this 👍 
We can also remove `fieldType` from `Field` as it was added alongside `TableCalculationField`

Reply:
Done here: https://github.com/lightdash/lightdash/pull/10062/commits/2f8ca5ed0574d660d829e607fe9a0b6c3eaeb9ce